scripts/src/models/orders.ts

```diff
@@ -325,9 +325,13 @@ function extendedOrder(origin: OrderOrigin): (typeof Order) & OrderFactory {
 	return Object.assign({}, Order, {
 		"new"(data?: DeepPartial<Order>, ...context: Array<DeepPartial<OrderContext>>): Order {
 			data = Object.assign(
-				{ nonce: Order.DEFAULT_NONCE },
 				data,
 				{ origin });
+
+			if (isNothing(data.nonce)) {
```

Comment:
I would check if the property exists vs checking it's value, eg `data.hasOwnProperty('nounce')` or `'nounce' in data` if you want to live on the edge (I think it'll be ok). (cc @doodyparizada)